quimb/evo.py

```diff
@@ -422,6 +436,35 @@ def int_step_callback(t, y, H):
                 step_callback(t, pt, H)
 
         self._step_callback = step_callback
+
+        # if there is an integration stopping condition, set up the function
+        #   to use this
+
+        if int_stop is not None:
+            int_stop_no_tryexcept = int_stop
+
+            # try just passing time and state to a single callback function
+            # if it fails, pass time, state and hamiltonian
+            def int_stop(t, y, H):
+                try:
+                    int_stop_output = int_stop_no_tryexcept(t, y)
+                except TypeError as e:
+                    if 'positional' in e.args[0]:
+                        int_stop_output = int_stop_no_tryexcept(t, y, H)
+                    else:
+                        raise
+                return int_stop_output
+
+            if int_step_callback is not None:
+                int_step_callback_bare = int_step_callback
```

Comment:
I got a bit confused by all the different clauses and renaming, might be worth adding a comment along the lines of 'need to combine different compute and stop callbacks'.

Reply:
I've ended up restructuring _setup_callbacks quite considerably